src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -210,6 +214,22 @@ public AnomalyDetectionIndices(
 
         this.maxUpdateRunningTimes = maxUpdateRunningTimes;
         this.updateRunningTimes = 0;
+
+        this.AD_RESULT_FIELD_CONFIGS = null;
+    }
+
+    private void initResultMapping() throws IOException {
+        if (AD_RESULT_FIELD_CONFIGS != null) {
+            // we have already initiated the field
+            return;
+        }
+        String resultMapping = getAnomalyResultMappings();
+
+        Map<String, Object> asMap = XContentHelper.convertToMap(new BytesArray(resultMapping), false, XContentType.JSON).v2();
+        Object properties = asMap.get(CommonName.PROPERTIES);
+        if (properties instanceof Map) {
+            AD_RESULT_FIELD_CONFIGS = (Map<String, Object>) properties;
+        }
```

Comment:
If properties is not Map, that means the result index mapping JSON file is wrong, then we should throw exception?

Reply:
the exception will be caught by the caller's catch method and log the error and return false.  How about log an error if the index mapping json file is wrong to save another round of exception creation/caught?

Reply:
I'm ok to just log error and let the caller to throw exception, make sure the error message make sense to the user.

Reply:
as discussed offline, I logged the error in the initResultMapping method.